### PR TITLE
Update Unique Binary Search Tree II.java

### PR DESCRIPTION
**Pull request by ghost**
it is ok for "generate" to handle n<0;

### CODE REVIEW

File: Java/Unique Binary Search Tree II.java [hunk 1]
```diff
@@ -45,10 +45,10 @@
  */
 public class Solution {
     public List<TreeNode> generateTrees(int n) {
-        List<TreeNode> rst = new ArrayList<>();
-        if (n <= 0) {
-            return rst;
-        }
+     //   List<TreeNode> rst = new ArrayList<>();
+     //   if (n <= 0) {
+     //      return rst;
```
Comment (awangdev) [inline Java/Unique Binary Search Tree II.java:50]: depending on test case. My assumption is to return `[]`, where in `generate`, it returns [null]. I think either way works.

**Comment by awangdev** [comment]
The original solution assume `[]` for input `n<=0`.  I'll not merge this since it's not a actual code fix.

**Comment by xianyuanxu** [comment]
get,and thinks!

**Comment by ghost** [comment]
get it,thanks!

**Pull request closed without merging**